src/dataExplorer/components/ResultsPane.tsx

```diff
@@ -119,9 +119,33 @@ const ResultsPane: FC = () => {
   }
 
   const downloadByServiceWorker = () => {
-    // TODO (chunchun): https://github.com/influxdata/ui/issues/6704
     if (language === LanguageType.INFLUXQL) {
-      console.error('csv download for InfluxQL is not implemented')
+      try {
+        event('runQuery.downloadCSV.influxql', {context: 'query experience'})
+
+        const params = new URLSearchParams({
+          db: selection.dbrp?.database,
+          q: text,
+        })
+        if (selection.dbrp?.retention_policy) {
+          params.set('rp', selection.dbrp?.retention_policy)
+        }
+        const url = `${API_BASE_PATH}query?${params}`
+        const hiddenForm = document.createElement('form')
+        hiddenForm.setAttribute('id', 'downloadDiv')
+        hiddenForm.setAttribute('style', 'display: none;')
+        hiddenForm.setAttribute('method', 'post')
+        hiddenForm.setAttribute('action', url)
```

Comment:
this is a duplicate of lines 155-160, perhaps you could think about making a reusable function

it also could be nice to pull out influxql vs flux/sql into functions that are just being called here based on language type rather than defined inline

Reply:
Sounds good. Updated in https://github.com/influxdata/ui/pull/6727/commits/4c8909596602e1a2f72ac28455397b797536dbd2